--- a/src/util/geo/Geo.tpp
+++ b/src/util/geo/Geo.tpp
@@ -6152,10 +6152,10 @@
   auto pa = centroid(a);
   auto pb = centroid(b);

-  double fA = webMercDistFactor(pa);
-  double fB = webMercDistFactor(pb);
+  double latA = 2 * atan(exp(pa.getY() / EQUATORIAL_RAD)) - 1.5707965;
+  double latB = 2 * atan(exp(pb.getY() / EQUATORIAL_RAD)) - 1.5707965;

-  return util::geo::dist(a, b) * cos((fA + fB) / 2.0);
+  return util::geo::dist(a, b) * cos((latA + latB) / 2.0);
 }

 // _____________________________________________________________________________
